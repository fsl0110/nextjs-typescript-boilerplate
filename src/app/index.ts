export * from '../tests/utils/testingLibraryUtils';
