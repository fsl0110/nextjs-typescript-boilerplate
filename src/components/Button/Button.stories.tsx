import { Story, Meta } from '@storybook/react';
import markdown from './Button.md';
import { Button, ButtonProps } from './Button';

export default {
  title: 'Example/Button',
  component: Button,
  parameters: {
    notes: markdown,
  },
} as Meta;

const Template: Story<ButtonProps> = (args) => (
  <Button {...args}>Button</Button>
);

export const Default = Template.bind({});
Default.args = {};

export const Small = Template.bind({});
Small.args = {
  size: 'sm',
};

export const Medium = Template.bind({});
Medium.args = {
  size: 'md',
};

export const Large = Template.bind({});
Large.args = {
  size: 'lg',
};

export const Gray = Template.bind({});
Gray.args = {
  color: 'gray',
};

export const Blue = Template.bind({});
Blue.args = {
  color: 'blue',
};

export const Green = Template.bind({});
Green.args = {
  color: 'green',
};
