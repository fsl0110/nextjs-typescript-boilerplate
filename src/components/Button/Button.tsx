import classNames from 'classnames';
import { FC, HTMLAttributes } from 'react';

// Storybook error: var trimmedText = (tag.text || '').trim();
// occurs when using @ in comments or using HTMLAttributes which uses @ in comments
// https://github.com/styleguidist/react-docgen-typescript/issues/356
// switched to typescript@4.2.4

export interface ButtonProps extends HTMLAttributes<HTMLButtonElement> {
  /**
   * What background color to use?
   *
   */
  color?: 'gray' | 'blue' | 'green';
  /**
   * How large should the button be?
   *
   */
  size?: 'sm' | 'md' | 'lg';
  /**
   * Optional click handler
   */
  onClick?: () => void;
}

export const Button: FC<ButtonProps> = ({
  color = 'blue',
  size = 'md',
  className,
  children,
  ...props
}) => {
  const styles = classNames({
    'bg-gray-400': color === 'gray',
    'bg-blue-400': color === 'blue',
    'bg-green-400': color === 'green',
    'text-base px-2.5 py-1': size === 'sm',
    'text-xl px-4 py-1.5': size === 'md',
    'text-3xl px-6 py-2.5': size === 'lg',
  });

  return (
    <button
      type="button"
      className={`text-white rounded-md ${styles} ${className}`}
      {...props}
    >
      {children}
    </button>
  );
};
