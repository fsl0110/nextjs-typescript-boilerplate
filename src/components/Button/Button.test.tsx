import { render } from 'app';
import { Button } from './Button';

describe('Button', () => {
  it('renders Button component', () => {
    render(<Button>Button</Button>);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Button>Button</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
