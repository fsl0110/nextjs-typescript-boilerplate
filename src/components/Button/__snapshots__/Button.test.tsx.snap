// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="text-white rounded-md bg-blue-400 text-xl px-4 py-1.5 undefined"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;
